vyper/codegen/core.py

```diff
@@ -923,6 +923,14 @@ def is_return_from_function(node):
         return True
     if isinstance(node, (vy_ast.Return, vy_ast.Raise)):
         return True
+    if isinstance(node, vy_ast.If):
+        return_if_body = is_return_from_function(node.body[0])
+        if node.orelse and is_return_from_function(node.orelse[0]):
+            return_else_body = True
+        else:
+            return_else_body = False
```

Comment:
```suggestion
        return_else_body = node.orelse and is_return_from_function(node.orelse[0])
```

Reply:
Actually, that's my first code.

https://github.com/vyperlang/vyper/pull/3273/commits/17c64751fed80396fd2c9da8f8f0ade3d15267c3

I was afraid that this code is trying to be smart. Okay, I'll update the code to the suggested one.

Reply:
it's a bit easier to follow I think